compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPUCheckLinalgVectorized.cpp

```diff
@@ -0,0 +1,38 @@
+// Copyright 2022 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/PassDetail.h"
+#include "iree/compiler/Codegen/Passes.h"
+#include "mlir/Dialect/Linalg/Utils/Utils.h"
+#include "mlir/Pass/Pass.h"
+
+namespace mlir {
+namespace iree_compiler {
+
+namespace {
+struct LLVMCPUCheckLinalgVectorizedPass
+    : LLVMCPUCheckLinalgVectorizedBase<LLVMCPUCheckLinalgVectorizedPass> {
+  void runOnOperation() override;
+};
+}  // namespace
+
+void LLVMCPUCheckLinalgVectorizedPass::runOnOperation() {
+  auto funcOp = getOperation();
+  auto walkResult = funcOp.walk([&](linalg::LinalgOp op) -> WalkResult {
+    return WalkResult::interrupt();
```

Comment:
Instead of interrupting the walk, would it make sense to just emit the warning from the LinalgOp and continue the walk? I think it could be very useful to get the full list of specific LinalgOps that were not vectorized.

Reply:
I was considering to emit ops or the function. Now I think emitting all the non-vectorized ops sounds good to me.

Reply:
It sounds good. Another option might be to include the function name and location in the error message of the LinalgOp. That would be useful to quickly filter or count missing vectorized ops per function with some linux commands.